cpp/src/io/csv/datetime.cuh

```diff
@@ -214,21 +133,18 @@ __inline__ __device__ bool extract_date(
  * digits. 12-hr and 24-hr time format is detected via the absence or presence of AM/PM characters
  * at the end.
  *
- * @param[in] begin Pointer to the first element of the string
- * @param[in] end Pointer to the first element after the string
- * @param[out] hour The hour value
- * @param[out] minute The minute value
- * @param[out] second The second value (0 if not present)
- * @param[out] millisecond The millisecond (0 if not present)
+ * @param begin Pointer to the first element of the string
+ * @param end Pointer to the first element after the string
+ * @return Duration in cudf milliseconds by summing up the extracted hours, minutes, seconds and
+ * milliseconds
  */
-__inline__ __device__ void extract_time(
-  char const* begin, char const* end, int* hour, int* minute, int* second, int* millisecond)
+__inline__ __device__ cudf::duration_ms extract_time(char const* begin, char const* end)
```

Comment:
@davidwendt same question here

Reply:
The string to duration code is defined in https://github.com/rapidsai/cudf/blob/branch-21.12/cpp/src/strings/convert/convert_durations.cu